File: Backend/api/src/modules/questions/application/use-cases/get-all-questions.use-case.ts
import { Injectable, Inject } from '@nestjs/common';
import { Question } from '../../domain/entities/question.entity';
import { QUESTION_REPOSITORY } from '../../domain/repositories/question.repository.interface';
import type { IQuestionRepository } from '../../domain/repositories/question.repository.interface';

@Injectable()
export class GetAllQuestionsUseCase {
  constructor(
    @Inject(QUESTION_REPOSITORY)
    private readonly questionRepo: IQuestionRepository,
  ) {}

  async execute(): Promise<Question[]> {
    return this.questionRepo.findAll();
  }
}

File: Movil/app/src/main/java/com/duelmath/features/questions/domain/usecases/GetAllQuestionsUseCase.kt
package com.duelmath.features.questions.domain.usecases

import com.duelmath.features.questions.domain.entities.Question
import com.duelmath.features.questions.domain.repositories.QuestionsRepository
import javax.inject.Inject

class GetAllQuestionsUseCase @Inject constructor(
    private val repository: QuestionsRepository,
) {
    suspend operator fun invoke(): Result<List<Question>> {
        return repository.getAllQuestions()
    }
}